Gems/Terrain/Code/Source/TerrainRaycast/TerrainRaycastContext.cpp

```diff
@@ -210,7 +210,7 @@ namespace
     // |    |    |    |    |    |    |    |    |
     // |____|____|____|____|____|____|____|____|
     inline void FindNearestIntersectionIterative(const TerrainSystem& terrainSystem,
-                                                 const AZ::Vector2& terrainResolution,
+                                                 float terrainResolution,
```

Comment:
If it's all the same I think it's preferable to leave this API the same and have the calling code pass in a AZ::Vector2 where both components are the same. We're unlikely to ever need it, but it's possible we might want this to work with non-square resolutions in the future for some reason.

Reply:
The opposite could also be true though - in the future we may want to refactor ray casts in such a way where they can be more efficient if the x/y step size is the same. Keeping the API the way it is forces us to always promise that functionality even if we don't actually need it now.

Reply:
Would love a third opinion on this - @mbalfour-amzn ?

Reply:
Optimizing for the square resolution case could be done within the function itself though, without changing the API. If the optimized implementation ends up being vastly different then we could create a separate version of the function which takes a single float, and the existing function can just call that if terrainResolution.GetX() == terrainResolution.GetY(), otherwise falling through to the current implementation in order to retain the ability to ray cast against terrain with a non-square resolution.

Reply:
Honestly looking at the API again I kinda think we shouldn't be passing in the query resolution and world bounds at all - that's information that you can already get from the TerrainSystem. However I also don't care that much and don't want to hold the review up so I went ahead and switched FindNearestIntersectionIterative back to the way it was. RayIntersect however needs to construct the Vector2 from a float value that it gets from TerrainSystem so I still am not sure how this benefits anyone. It's a private function that we're sending extra data to just to ignore it.